c++/src/linked-list.cc

```diff
@@ -174,31 +174,23 @@ class LinkedList {
     }
 
     Node<T>* lookingAt = this->head;
-    Node<T>* nodeToDelete = nullptr;
     Node<T>* previousNode = nullptr;
 
-    while (true) {
+    while (lookingAt->getValue() != value) {
       // find the node containing the value to delete
-      
-      if (lookingAt == nullptr) {return;}
-      else if (lookingAt->getValue() == value) {
-        nodeToDelete = lookingAt;
-        break;
-      }
-
       previousNode = lookingAt;
       lookingAt = lookingAt->getNext();
-    }
 
-    if (nodeToDelete) {
-      if (previousNode) {
-        previousNode->setNext(nodeToDelete->getNext());
-      } else {
-        this->head = nodeToDelete->getNext();
-      }
+      if (lookingAt == nullptr) {return;}
```

Comment:
Because the way you're doing it now you still end up duplicating this check here, even though with the change I described this wouldn't be necessary, unless I'm missing something?